language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/AnnotationNodeContext.java

```diff
@@ -71,53 +71,112 @@ public AnnotationNodeContext() {
+                && !currentModule.name.value.equals(annotationOwner.name.value) && !("ballerina".equals(orgName)
+                && "lang.annotations".equals(value));
+    }
+
+    private List<LSCompletionItem> getCurrentModuleAnnotations(LSContext ctx, SyntaxKind kind,
+                                                               Map<String, String> pkgAliasMap) {
         BLangPackage bLangPackage = ctx.get(DocumentServiceKeys.CURRENT_BLANG_PACKAGE_CONTEXT_KEY);
-        List<LSCompletionItem> completionItems = new ArrayList<>();
-        List<BLangAnnotation> annotations = bLangPackage.topLevelNodes.stream()
+        List<BAnnotationSymbol> annotations = bLangPackage.topLevelNodes.stream()
                 .filter(topLevelNode -> topLevelNode instanceof BLangAnnotation)
-                .map(topLevelNode -> (BLangAnnotation) topLevelNode)
+                .map(topLevelNode -> (BAnnotationSymbol) ((BLangAnnotation) topLevelNode).symbol)
+                .collect(Collectors.toList());
+
+        return this.getAnnotationCompletionsForSymbols(ctx, annotations, kind, pkgAliasMap);
+    }
+
+    private List<LSCompletionItem> getAnnotationsInModule(LSContext context, String alias, SyntaxKind kind,
+                                                          Map<String, String> pkgAliasMap) {
+        Optional<Scope.ScopeEntry> moduleEntry = CommonUtil.packageSymbolFromAlias(context, alias);
+        if (!moduleEntry.isPresent()) {
+            List<LSCompletionItem> completionItems = new ArrayList<>();
+            // Import statement has not been added. Hence try resolving from the annotation cache
+            LSAnnotationCache.getInstance().getAnnotationsInModule(context, alias, kind)
+                    .forEach((key, value) -> value.forEach(annotation -> {
+                        completionItems.add(CommonUtil.getAnnotationCompletionItem(key, annotation, context, false,
+                                pkgAliasMap));
+                    }));
+
+            return completionItems;
+        }
+
+        List<BAnnotationSymbol> collect = moduleEntry.get().symbol.scope.entries.values().stream()
+                .filter(entry -> {
+                    BSymbol symbol = entry.symbol;
+                    return symbol instanceof BAnnotationSymbol && ((symbol.flags & Flags.PUBLIC) == Flags.PUBLIC);
+                })
+                .map(scopeEntry -> (BAnnotationSymbol) scopeEntry.symbol)
                 .collect(Collectors.toList());
-        BLangNode scopeNode = ctx.get(CompletionKeys.SCOPE_NODE_KEY);
 
-        annotations.forEach(bLangAnnotation -> {
-            BAnnotationSymbol symbol = (BAnnotationSymbol) bLangAnnotation.symbol;
+        return this.getAnnotationCompletionsForSymbols(context, collect, kind, pkgAliasMap);
+    }
+
+    private SyntaxKind getParentSyntaxKind(AnnotationNode node) {
+        if (node.parent().kind() == SyntaxKind.METADATA) {
```

Comment:
Can we move `node.parent().kind()` into local variable and Line 140-144 can be simplified into a ternary return statement.